Missing Constructor in Plugin Java Class

### Environment

node version: 20.0.10


### Is used with npx or as a local dependency?

- [X] npx
- [ ] local dependency

### Affected commands

- [X] android
- [ ] ios

### Current behavior

`npx expo run:android` fails to build with java build errors looking like this
```
receipt-scanner-rush/android/app/src/main/java/com/lordxerus/receiptscannerrush/receiptocrframeprocessorplugin/ReceiptOCRFrameProcessorPluginPluginPackage.java:19: error: constructor ReceiptOCRFrameProcessorPluginPlugin in class ReceiptOCRFrameProcessorPluginPlugin cannot be applied to given types;
            options -> new ReceiptOCRFrameProcessorPluginPlugin(options)
                       ^
  required: no arguments
  found:    Map<String,Object>
  reason: actual and formal argument lists differ in length
receipt-scanner-rush/android/app/src/main/java/com/lordxerus/receiptscannerrush/receiptocrframeprocessorplugin/ReceiptOCRFrameProcessorPluginPlugin.java:9: error: constructor FrameProcessorPlugin in class FrameProcessorPlugin cannot be applied to given types;
public class ReceiptOCRFrameProcessorPluginPlugin extends FrameProcessorPlugin {
       ^
  required: Map<String,Object>
  found:    no arguments
  reason: actual and formal argument lists differ in length
```
**Actual and formal argument lists differ in length**
_pls ignore my interesting names im new to dis_

I believe this is because the default code in `android-utils.ts` does not generate a constructor
As a result, Java generates a default constructor with no parameters and calls the super constructor with no arguments
Which I believe causes the errors above.

Reproduction Instructions:
1. Create Java Plugin
2. No Constructor is generated
3. Gradle build does not error

I'm fairly sure my conclusion is correct so I will provide a reproduction repository upon request.

### Expected behavior

`npx expo run:android` builds successfully

My solution to the above errors is to add a dummy constructor, maybe with comments

```java
ReceiptOCRFrameProcessorPluginPlugin(@Nullable Map<String, Object> options) {
    super(options);
    Log.d("ExamplePlugin", "ExampleFrameProcessorPlugin initialized with options: " + options);
  }
```

### Reproduction repository link

Reproduction repository upon request; see Current Behavior for repro

--- a/src/android-utils.ts
+++ b/src/android-utils.ts
@@ -46,7 +46,7 @@ export const extractPackageName = (workingDir: string, manifestPath: string) => 
  const results = [ ...groovyGradleResults, ...kotlinGradleResults ];
 
  for (const gradleScriptPath of results) {
-   const gradleScriptContent = fs.readFileSync(gradleScriptPath, { encoding: 'utf8' });
+   const gradleScriptContent = fs.readFileSync(gradleScriptPath, { encoding: 'utf-8' });
 
    const namespaceResults = gradleScriptContent.match(NAMESPACE_VALUE_REGEX);
 
@@ -140,6 +140,10 @@ import com.mrousavy.camera.frameprocessor.FrameProcessorPlugin;
 import java.util.Map;
 
 public class ${pluginName}Plugin extends FrameProcessorPlugin {
+  public ${pluginName}Plugin(@Nullable Map<String, Object> options) {
+    super(options);
+  }
+
   @Nullable
   @Override
   public Object callback(@NonNull Frame frame, @Nullable Map<String, Object> arguments) {
@@ -193,7 +197,6 @@ import com.facebook.react.ReactPackage;
 import com.facebook.react.bridge.NativeModule;
 import com.facebook.react.bridge.ReactApplicationContext;
 import com.facebook.react.uimanager.ViewManager;
-import com.mrousavy.camera.frameprocessor.FrameProcessorPlugin;
 import com.mrousavy.camera.frameprocessor.FrameProcessorPluginRegistry;
 ${isApplicationPackage ? '' : `import ${packageName}.${pluginName.toLowerCase()}.${pluginName}Plugin;\n`}
 import java.util.Collections;

--- a/src/android.ts
+++ b/src/android.ts
@@ -91,8 +91,8 @@ export async function androidCommandHandler(argv: Arguments<unknown>) {
 
   spinner.text = 'Extracting android package name';
   spinner.start();
-  
-  const packageName = extractPackageName(path.resolve(manifestPath, '..', '..'), manifestPath);
+
+  const packageName = extractPackageName(path.resolve(path.dirname(manifestPath), '..', '..'), manifestPath);
 
   if (!packageName) {
     spinner.fail();

--- a/src/ios-utils.ts
+++ b/src/ios-utils.ts
@@ -176,7 +176,7 @@ const createFileForIOSImplementation = (
   content: string,
 ) => {
   if (!fs.existsSync(filepath)) {
-    fs.writeFileSync(filepath, content, { encoding: 'utf8' });
+    fs.writeFileSync(filepath, content, { encoding: 'utf-8' });
   }
 };
 

